lib/spack/spack/util/path.py

```diff
@@ -27,6 +28,35 @@
     'tempdir': tempfile.gettempdir(),
 }
 
+# This is intended to be longer than the part of the install path
+# spack generates from the root path we give it.  Included in the
+# estimate:
+#
+#   os-arch      ->   30
+#   compiler     ->   30
+#   package name ->   50   (longest is currently 47 characters)
+#   version      ->   20
+#   hash         ->   32
+#   buffer       ->  138
+#  ---------------------
+#   total        ->  300
+SPACK_MAX_INSTALL_PATH_LENGTH = 300
+SPACK_PATH_PADDING_CHARS = 'spack_path_placeholder'
+SPACK_PATH_PADDING_SIZE = len(SPACK_PATH_PADDING_CHARS)
+
+try:
+    path_max_proc  = subprocess.Popen(['getconf', 'PATH_MAX', '/'],
```

Comment:
don't do this at module scope -- it means that this is run any time this file is imported (regardless of whether it's used), and enough of these will slow down Spack startup time.   Instead, make functions that get these values.